File: src/main/java/com/webshop/babunov/model/Item.java
package com.webshop.babunov.model;

import javax.persistence.*;
import javax.validation.constraints.NotNull;


@Entity
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @NotNull(message = "Item name is required.")
    @Basic(optional = false)
    private String name;

    private Double price;

    private String pictureUrl;

    public Item(Long id, @NotNull(message = "Item name is required.") String name, Double price, String pictureUrl) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.pictureUrl = pictureUrl;
    }

    public Item() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public String getPictureUrl() {
        return pictureUrl;
    }

    public void setPictureUrl(String pictureUrl) {
        this.pictureUrl = pictureUrl;
    }

}

File: web-front/src/app/models/item.ts
export class Item {
    id: number;
    name: string;
    pictureUrl: string;
    price: number;
    description: string;

    constructor(id: number, name: string, price: number = 0, description : string = "Описание отсутствует", pictureUrl: string = "https://i.imgur.com/8UdKNS4.jpeg") {
        this.id = id;
        this.name = name;
        this.pictureUrl = pictureUrl;
        this.price = price;
        this.description = description;
    }
}
